src/Sql/dbo/Stored Procedures/User_BumpAccountRevisionDateByOrganizationUserIds.sql

```diff
@@ -0,0 +1,27 @@
+CREATE PROCEDURE [dbo].[User_BumpAccountRevisionDateByOrganizationUserIds]
+    @OrganizationUserIds [dbo].[GuidIdArray] READONLY
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    SELECT
+        UserId
+    INTO
+        #UserIds
+    FROM
+        [dbo].[OrganizationUser] OU
+    INNER JOIN
+        @OrganizationUserIds OUIds on OUIds.Id = OU.Id
+    WHERE
+        OU.[Status] = 2 -- Confirmed
+
+    UPDATE
+        U
+    SET
+        U.[AccountRevisionDate] = GETUTCDATE()
+    FROM
+        [dbo].[User] U
+    Inner JOIN
```

Comment:
Curious, why not just use an `INNER JOIN` (casing btw) to `[dbo].[OrganizationUser] OU` instead of a temp table here? Temp tables via `tempdb` can be expensive, especially for smaller amounts of data, and in this case, you're already querying the `[dbo].[OrganizationUser]` table directly based on `Id` (it's clustered index) so that's already pretty fast, using `INNER JOIN` criteria on `OUIds.[Id] = OU.[Id] AND OU.[Status] = 2 AND U.[Id] = OU.[UserId]` along those lines should be efficient enough for a single set-based operation (vs. 2 operations using a temp table).

Reply:
hmm. I didn't think of this. Do you mean something like this? 

```
    UPDATE
        U
    SET
        U.[AccountRevisionDate] = GETUTCDATE()
    FROM
        @OrganizationUserIds OUIDs
    INNER JOIN
        [dbo].[OrganizationUser] OU ON OUIDs.Id = OU.Id
    INNER JOIN
        [dbo].[User] U ON OU.UserId = U.Id
    WHERE
        OU.[Status] = 2 -- Confirmed
```

Or do you see a way to do it with one join?

Reply:
Yeah, pretty much exactly that, you'll need 2 joins to do it, but one being an in-memory table and the other being a table on it's underlying PK and clustered index, it should be pretty fast as a single, set-based operation.

Reply:
Oh, and you can remove the `WHERE` clause and just put `OU.[Status] = 2 -- Confirmed` as an `AND` condition for the `INNER JOIN` to `OU` itself.

```sql
    UPDATE
        U
    SET
        U.[AccountRevisionDate] = GETUTCDATE()
    FROM
        @OrganizationUserIds OUIDs
    INNER JOIN
        [dbo].[OrganizationUser] OU ON OUIDs.Id = OU.Id
        AND OU.[Status] = 2 -- Confirmed
    INNER JOIN
        [dbo].[User] U ON OU.UserId = U.Id
```

Reply:
nice -- thanks!

I wasn't sure you could join on a constant value like that, cool.